src/Nethermind/Nethermind.Config/ConfigProvider.cs

```diff
@@ -86,14 +88,32 @@ public void AddSource(IConfigSource configSource)
         _configSource.Add(configSource);
     }
 
-    public void Initialize()
+    private static IList<(Type ConfigType, Type DirectImplementation)> GatherConfigsFromAssembly()
     {
+        if (_configTypesCache is not null) return _configTypesCache;
```

Comment:
Can you make it to self-initialized property that calls this method internally? IMO it is more readable and standard.